Simplenote/Classes/SPResultsController.swift

```diff
@@ -0,0 +1,150 @@
+import Foundation
+import CoreData
+
+
+// MARK: - SPResultsController
+//
+class SPResultsController {
+
+    /// Batch Size for the FRC's Request
+    ///
+    private let resultsBatchSize = 20
+
+    /// Fetch Request: Note
+    ///
+    private lazy var request: NSFetchRequest<Note> = {
+        let request = NSFetchRequest<Note>()
+        request.entity = NSEntityDescription.entity(forEntityName: Note.classNameWithoutNamespaces, in: mainContext)
+        request.fetchBatchSize = resultsBatchSize
+        request.sortDescriptors = sortDescriptors()
+        return request
+    }()
+
+    /// Results Controller: Note Entity
+    ///
+    private lazy var resultsController: NSFetchedResultsController<Note> = {
+        NSFetchedResultsController<Note>(fetchRequest: request, managedObjectContext: mainContext, sectionNameKeyPath: nil, cacheName: nil)
+    }()
+
+    /// Main MOC
+    ///
+    private let mainContext: NSManagedObjectContext
+
+    /// Keyword we should be filtering by
+    ///
+    var keyword = String() {
+        didSet {
+            guard oldValue != keyword else {
+                return
+            }
+
+            refreshPredicate(keyword: keyword)
+        }
+    }
+
+    /// Designated Initializer
+    ///
+    ///  - mainContext: Main Thread's MOC
+    ///
+    init(mainContext: NSManagedObjectContext) {
+        self.mainContext = mainContext
```

Comment:
Maybe consider an assert to confirm the passed context has main queue concurrency?

Reply:
Incidentally, tho I'm a fan of the dependency injection here I'm wondering if its not more straight forward to just hard code `SPAppDelegate.shared().managedObjectContext` since that's what's ultimately used anyway. Is there a scenario where there are different main queue contexts in play, or a test scenario were a different context would be used?

Reply:
Thank you Eric!! I ended up doing as follows, so that it's flexible enough for DI, but you can't really break things up.

Eventually I'd love to write a million unit tests.

```
init(mainContext: NSManagedObjectContext = SPAppDelegate.shared().managedObjectContext) {
        assert(mainContext.concurrencyType == .mainQueueConcurrencyType)
```